src/protocols/rdp/rdp.h

```diff
@@ -153,6 +168,29 @@ typedef struct guac_rdp_client {
      */
     guac_common_list* available_svc;
 
+    /**
+     * Lock which is locked when one or more credentials are required to
+     * complete the connection, and unlocked when credentials have been
+     * provided by the client.
+     */
+    pthread_mutex_t rdp_credential_lock;
+    
+    /**
+     * Condition which is used when the pthread needs to wait for one or more
+     * credentials to be provided by the client.
+     */
+    pthread_cond_t rdp_credential_cond;
+    
+    /**
+     * Flags for tracking events related to the rdp_credential_cond
+     * pthread condition.
```

Comment:
As with VNC, the nature of this value with respect to those flags should be documented here.